bigtop-tests/smoke-tests/hive/HiveJdbcGeneralTest.java

```diff
@@ -0,0 +1,390 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+import org.apache.hive.service.cli.HiveSQLException;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ *
+ * THIS CLASS TESTS THE FOLLOWING ASPECTS OF HIVE:
+ *
+ * Connect to hive; Drop the tables if already there; Test Show Tables; Create
+ * blank tables, both transactional and non, along with ORC format and
+ * partitions; Reprint the list of tables; Load File into HDFS; Load data into
+ * tables; describe tables; Delete Uploaded File; Print table contents with
+ * various queries; Test Prepared and Callable Statements; Test Fetch Sizes;
+ * Test ACID (UPDATE/INSERT/DELETE); View Partitions;
+ */
+public class HiveJdbcGeneralTest extends TestMethods {
+
+  static String hivePort = System.getenv("HIVE_PORT");
+  static String jdbcConnection = System.getenv("HIVE_JDBC_URL");
+  static Connection con;
+  String newTableName = "btest";
+
+  @BeforeClass
+  public static void onTimeSetup() throws Exception {
+    String username = System.getenv("HIVE_USER");
+    String password = System.getenv("HIVE_PASSWORD");
+    Properties connectionProps = new Properties();
+    connectionProps.put("user", username);
+    connectionProps.put("password", password);
+    Class.forName("org.apache.hive.jdbc.HiveDriver");
+    con = DriverManager.getConnection(
+        jdbcConnection + ":" + hivePort + "/default;", connectionProps);
+  }
+
+  @AfterClass
+  public static void oneTimeTearDown() throws Exception {
+    con.close();
+  }
+
+  @Test // (expected=java.sql.SQLDataException.class)
+  public void testTableCreation() throws Exception {
+    final File f = new File(HiveJdbcGeneralTest.class.getProtectionDomain()
+        .getCodeSource().getLocation().getPath());
+    String hdfsConnection =
+        propertyValue("hdfs-site.xml", "dfs.namenode.rpc-address");
+    try (Statement stmt = con.createStatement()) {
+      String columnNames =
+          "(Flight int, Dates varchar(255), Depart varchar(10), Orig varchar(5), Dest varchar(5), Miles int, Boarded int, Capacity double)";
+      String partitionedColumns =
+          "(Flight int, Dates varchar(255), Depart varchar(10), Orig varchar(5), Dest varchar(5), Miles int, Boarded int)";
+      String localFilepath = f + "/samdat1.csv";
+      String HdfsURI = "hdfs://" + hdfsConnection;
+      String filePath = "/tmp/htest/00000_";
+      String fileDestination = HdfsURI + filePath;
+      assertFalse(con.getMetaData().supportsRefCursors());
+      assertTrue(con.getMetaData().allTablesAreSelectable());
+      assertEquals("Apache Hive", con.getMetaData().getDatabaseProductName());

```

Comment:
It's OK to assert versioning and configurations. Just move these out from testTableCreation to another test and name it clearly.